### PR TITLE
Silence messages from libbpf

### PR DESCRIPTION
**Pull request by keszybz**
For https://bugzilla.redhat.com/show_bug.cgi?id=2084955#c14. ~~It probably doesn't fix the issue completely, because there should be still one more message `Failed to load BPF object`.~~

### CODE REVIEW

**Comment by keszybz** [comment]
Seems to work fine.

File: src/core/bpf-util.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        r = cg_unified_controller(SYSTEMD_CGROUP_CONTROLLER);
+        if (r < 0) {
+                log_warning_errno(r, "Can't determine whether the unified hierarchy is used: %m");
+                return (supported = false);
+        }
+
+        if (r == 0) {
+                log_info_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
+                               "Not running with unified cgroup hierarchy, BPF cgroup filters are not supported");
```
Comment (daandemeyer) [inline src/core/bpf-util.c]: Copy paste error with the log messages?

File: src/core/bpf-util.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                return (supported = false);
+        }
+
+        r = dlopen_bpf();
+        if (r < 0) {
+                log_info_errno(r, "Failed to open libbpf, BPF cgroup filters disabled: %m");
```
Comment (daandemeyer) [inline src/core/bpf-util.c]: Ditto

File: src/core/bpf-util.c (hunk 1 above)
Comment (keszybz) [reply src/core/bpf-util.c]: Hmm, can you be more specific?

Comment (daandemeyer) [reply src/core/bpf-util.c]: > BPF cgroup filters are not supported

This looked specific for one use case of BPF but I might understand BPF wrong

Comment (keszybz) [reply src/core/bpf-util.c]: "BPF cgroup filters" was supposed to be a generic term that covers the types of filters we attach… I think it's OK,  but suggestions are welcome.

Comment (daandemeyer) [reply src/core/bpf-util.c]: I got tripped up by the function being named bpf_supported(). I figure we might use bpf for something different than cgroup filters in the future and then the message would be wrong if we also use bpf_supported() there to check for bpf support. 

So I'd say either the function should be renamed to something more specific to indicate it's checking more than just whether bpf is supported or the error message should be updated to say we don't support any BPF features unless we're using the unified hierarchy, so something like "Not running with unified cgroup hierarchy, disabling BPF features"

Comment (keszybz) [reply src/core/bpf-util.c]: I renamed the function to `cgroup_bpf_supported()` and adjusted the texts a bit.

**Review by yuwata — APPROVED** [review]

**Pull request merged**